common/src/View/CameraTool3D.cpp

```diff
@@ -88,6 +88,10 @@ namespace TrenchBroom {
                 const auto maxDistance = std::max(vm::intersect_ray_plane(m_camera.viewRay(), orbitPlane) - 32.0f, 0.0f);
                 const auto distance = std::min(factor * scrollDist * moveSpeed(false), maxDistance);
                 m_camera.moveBy(distance * m_camera.direction());
+            } else if (adjustFlySpeed(inputState)) {
+                const auto speed = pref(Preferences::CameraFlyMoveSpeed);
+                const auto newSpeed = vm::clamp(speed + factor * (scrollDist > 0.0f ? 0.1f : -0.1f), 0.0f, 2.0f);
+                setPref(Preferences::CameraFlyMoveSpeed, newSpeed);
```

Comment:
Comments about `setPref`:
- Doing the mousewheel fly speed adjustment causes hitches on Debug builds (at least on Windows). Haven't checked if it's due to the I/O or the notifiers running. It doesn't hitch (or at least, much less) on RelWithDebInfo builds.
- I get no hitches if I do:

```
PreferenceManager& prefs = PreferenceManager::instance();
prefs.set(Preferences::CameraFlyMoveSpeed, newSpeed);
```

although this isn't a workable solution since it never saves.
I think the ideal solution would be if we would only save the prefs after the RMB is released (and only if the prefs were actually changed by using the mousewheel).

- I discovered few other issues while testing this. I had the `Preferences.json` open in VS code. Was testing the mousewheel speed adjustment in TB and after 10-15 seconds, I got the "failed to write preference file" dialog from TB (the one shown in `PreferenceManager::saveChanges()` if writing fails.) I'm not sure why writing failed.. maybe somehow having the prefs file open in VS code at the time caused it, maybe a Windows thing. It's reproducible, though. The problem was, once I dismissed this dialog box, TB rendering was broken (black triangle covering half of the map views.) It makes sense that showing a modal dialog during a `RenderView::paintGL` call will break the GL state, but it does mean we need to be more careful about what code runs within `paintGL`.


I'd say the hitching is acceptable on RelWithDebInfo at least for a first pass of this. The "failed to save" thing, I will move to a new github issue.

Reply:
Ok, I have made it so that it only saves on RMB up, and I have added a change that will only save the preferences when there actually are unsaved preferences at all.